infer/modules/train/extract_feature_print.py

```diff
@@ -91,8 +93,9 @@ def readwave(wav_path, normalize=False):
 model = models[0]
 model = model.to(device)
 printt("move model to %s" % device)
-if device not in ["mps", "cpu"]:
-    model = model.half()
+if is_half:
+    if device not in ["mps", "cpu"]:
+        model = model.half()

```

Comment:
mps and cpu need to use half

Reply:
I don't quite understand why this is the case
Judging from how config.py is written mps and cpu need to *not* use half

https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/0de65053247fb616fa696e4c40d32411174b6310/configs/config.py#L168-L177

I have no experience working with mps though

Reply:
Oh really, What was I thinking?
I rewroted this config.py and also the configuration for mps